### PR TITLE
Affiliates disclaimer

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Updates the 'Affiliates disclaimer' placement as follows:

- until leftcol: inline alongside the second paragraph
- from leftcol: below the meta section

### Until leftcol

| 320px       | 740px      | 980px      |
| ----------- | ---------- | -- |
|  |  |  |

### From leftcol

|  1140px      |
|  -- |
| Screenshot 2024-03-13 at 16 29 15 |

### CODE REVIEW

File: dotcom-rendering/src/lib/article.ts [hunk 1]
```diff
@@ -39,6 +40,7 @@ export const enhanceArticleType = (
 		renderingTarget,
 		promotedNewsletter: data.promotedNewsletter,
 		imagesForLightbox,
+		hasAffiliateLinksDisclaimer: !!data.affiliateLinksDisclaimer,
```
Comment (arelra) [inline dotcom-rendering/src/lib/article.ts:43]: The truthy value of `hasAffiliateLinksDisclaimer` in the incoming JSON from Frontend determines if we call `enhanceDisclaimer` in `enhanceBlocks`

File: dotcom-rendering/src/components/Disclaimer.tsx [hunk 2]
```diff
@@ -8,20 +10,56 @@ const disclaimerStyles = css`
 	sup {
 		font-size: 85%;
 	}
-
 	margin-bottom: 16px;
+	color: inherit;
+	text-decoration: none;
+`;
+
+const disclaimerInlineStyles = css`
```
Comment (arelra) [inline dotcom-rendering/src/components/Disclaimer.tsx:18]: This styling was ~lifted~ inspired by the styling of `RichLink.tsx`

Are there better css custom properties for a generic inline component?

**Review by Jakeii — APPROVED** [review]

File: dotcom-rendering/src/components/Disclaimer.tsx [hunk 3]
```diff
@@ -8,20 +10,56 @@ const disclaimerStyles = css`
...
+	padding-left: 5px;
+	padding-bottom: 5px;
 `;
 
-type Props = {
-	html: string;
-};
+const DisclaimerText = () => (
+	<p>
+		<sup>
```
Comment (domlander) [inline dotcom-rendering/src/components/Disclaimer.tsx:36]: Why is a `sup` element used here? It doesn't seem to fit with the usage notes or examples in the MDN docs: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/sup#usage_notes

Maybe an `aside` would be a more appropriate element for this block of text?

Comment (domlander) [reply dotcom-rendering/src/components/Disclaimer.tsx:36]: Ah, I see you're already using an aside in the `AffiliateDisclaimer` component. I guess my question would then be: is the `sup` element necessary?

**Comment by domlander** [comment]
The affiliate link is also at the bottom of the article content on screen sizes up to `leftCol`

image

**Comment by emma-imber** [comment]
> The affiliate link is also at the bottom of the article content on screen sizes up to `leftCol`
> 
> image

This is fixed in the corresponding frontend PR: https://github.com/guardian/frontend/pull/26969 The current frontend code sends a disclaimer block appended to the HTML but we've removed that now. The bottom disclaimer is not visible when testing the two branches together on CODE

**Review by domlander — APPROVED** [review]

File: dotcom-rendering/src/components/Disclaimer.tsx (hunk 2 above)
Comment (georgeblahblah) [reply dotcom-rendering/src/components/Disclaimer.tsx:18]: I don't think so but we could probably rename them to uncouple the var name a bit?

**Review by georgeblahblah — APPROVED** [review]

**Comment by paperboyo** [comment]
Text Sans: 👏 .

I’m no designer, but text looks tad tolouse given the reduced size in comparison to body? Shouldn’t we tighten line height? (and gain valuable space at the same time as making it look more contained)

File: dotcom-rendering/src/components/Disclaimer.tsx (hunk 3 above)
Comment (arelra) [reply dotcom-rendering/src/components/Disclaimer.tsx:36]: This was carried over from Frontend but good shout, I'll remove the sup.

**Comment by arelra** [comment]
> Text Sans: 👏 .
> 
> I’m no designer, but text looks tad tolouse given the reduced size in comparison to body? Shouldn’t we tighten line height? (and gain valuable space at the same time as making it look more contained)

Thanks. The urgency of this change meant we didn't get the normal level of design input but I'll follow up on your excellent suggestions.

**Comment by arelra** [comment]
@georgeblahblah @paperboyo @domlander Your suggestions applied here: https://github.com/guardian/dotcom-rendering/pull/10971

**Pull request merged**